tests/unit/remote/ssh/test_connection.py

```diff
@@ -0,0 +1,46 @@
+import mockssh
+import os
+from pytest import yield_fixture
+
+from dvc.remote.ssh import RemoteSSH
+
+__here__ = os.path.abspath(os.path.dirname(__file__))
+
+user = "user"
+key_path = os.path.join(__here__, "{0}.key".format(user))
+
+
+@yield_fixture()
+def server():
+    users = {user: key_path}
+    with mockssh.Server(users) as s:
+        yield s
+
+
+def _client(server):
+    url = "ssh://127.0.0.1:/path/to/dir"
+    config = {"url": url}
+    remote = RemoteSSH(None, config)
+    remote.keyfile = key_path
+    client = remote.ssh(host=remote.host, user=user, port=server.port)
```

Comment:
You are testing SSHConnection here, so you don't have to(and probably shouldn't) instantiate it through RemoteSSH, since you could instantiate SSHConnection directly.

Reply:
I updated the point. I'm new our community and need a little bit of time to fully understand source code base